makefile

```diff
@@ -96,9 +96,21 @@ htmlCoverageOutput := $(foreach target,$(testPackages),$(buildDir)/output.$(targ
 # end output files
 
 # start basic development targets
-proto:
-	@mkdir -p rpc/internal
-	protoc --go_out=plugins=grpc:rpc/internal *.proto
+protocVersion := 3.6.1
+protocGenGoVersion := 1.3.2
+protoOS := $(shell uname -s | tr A-Z a-z)
+ifeq ($(protoOS),darwin)
+protoOS := osx
+endif
+protoOS := $(protoOS)-$(shell uname -m | tr A-Z a-z)
+$(buildDir)/protoc:
+	curl --retry 10 --retry-max-time 60 -L0 https://github.com/protocolbuffers/protobuf/releases/download/v$(protocVersion)/protoc-$(protocVersion)-$(protoOS).zip --output protoc.zip
+	unzip -q protoc.zip -d $(buildDir)/protoc
+	rm -f protoc.zip
+	GOBIN=$(abspath $(buildDir)) $(gobin) install github.com/golang/protobuf/protoc-gen-go@v$(protocGenGoVersion)
+proto: $(buildDir)/protoc
+	mkdir -p rpc/internal
+	PATH=$(PATH):$(abspath $(buildDir)) $(buildDir)/protoc/bin/protoc --go_out=plugins=grpc:rpc/internal *.proto
```

Comment:
The PATH has to be `PATH="$(abspath $(buildDir)):$(PATH)"` or else any other `protoc-gen-go` or `protoc` executable already present in the `PATH` will be used. It still needs double quotes during assignment so that the entire `PATH` is considered as a single string by shell parsing rules.